src/test/java/org/opensearch/security/system_indices/SystemIndicesTests.java

```diff
@@ -326,27 +384,61 @@ public void testCloseOpen() throws Exception {
     }
 
     @Test
-    public void testCloseOpenWithSystemIndices() throws Exception {
+    public void testCloseOpenWithSystemIndicesShouldSucceedAsSuperAdmin() throws Exception {
         setupSystemIndicesEnabledWithSsl();
         createTestIndicesAndDocs();
         RestHelper keyStoreRestHelper = keyStoreRestHelper();
-        RestHelper sslRestHelper = sslRestHelper();
 
-        // as super-admin
         for (String index : listOfIndexesToTest) {
             RestHelper.HttpResponse responseClose = keyStoreRestHelper.executePostRequest(index + "/_close", "");
             assertEquals(RestStatus.OK.getStatus(), responseClose.getStatusCode());
+            assertTrue(responseClose.getBody().contains("{\"closed\":true}"));
 
             RestHelper.HttpResponse responseOpen = keyStoreRestHelper.executePostRequest(index + "/_open", "");
             assertEquals(RestStatus.OK.getStatus(), responseOpen.getStatusCode());
         }
 
-        // as admin
+    }
+
+    @Test
+    public void testCloseOpenIndexShouldFailWithSystemIndicesAsAdmin() throws Exception {
+        setupSystemIndicesEnabledWithSsl();
+        createTestIndicesAndDocs();
+        RestHelper sslRestHelper = sslRestHelper();
+
         for (String index : listOfIndexesToTest) {
             RestHelper.HttpResponse responseClose = sslRestHelper.executePostRequest(index + "/_close", "", allAccessUserHeader);
             assertEquals(RestStatus.FORBIDDEN.getStatus(), responseClose.getStatusCode());
+            Assert.assertTrue(
+                responseClose.getBody()
+                    .contains(
+                        "{\"type\":\"security_exception\",\"reason\":\"no permissions for [] and User [name=admin_all_access, backend_roles=[], requestedTenant=null]\"}"
+                    )
+            );
 
             RestHelper.HttpResponse responseOpen = sslRestHelper.executePostRequest(index + "/_open", "", allAccessUserHeader);
+            assertEquals(RestStatus.FORBIDDEN.getStatus(), responseOpen.getStatusCode());
+            Assert.assertTrue(
```

Comment:
Replace with `assertThat` using `containsString`, this will print the error message if it didn't match.

Alternatively, use `assertThat(responseOpen.getBody().findValueInJson(".reason"), equalTo("no permissions for [] and User [name=admin_all_access, backend_roles=[], requestedTenant=null]"))` so the test is more readable.

Note; occurs in several places, assertTrue/assertFalse are an anti-pattern

Reply:
**assertTrue/assertFalse are an anti-pattern**

 Very true! I think I was awake for too long when writing those tests  xD